src/cargo/ops/cargo_new.rs

```diff
@@ -62,19 +62,25 @@ pub struct NewOptions {
 pub enum NewProjectKind {
     Bin,
     Lib,
+    Auto,
 }
 
 impl NewProjectKind {
     fn is_bin(self) -> bool {
         self == NewProjectKind::Bin
     }
+
+    fn is_auto(self) -> bool {
+        self == NewProjectKind::Auto
+    }
 }
 
 impl fmt::Display for NewProjectKind {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         match *self {
             NewProjectKind::Bin => "binary (application)",
             NewProjectKind::Lib => "library",
+            NewProjectKind::Auto => "auto-select type",
```

Comment:
`NewProjectKind::Auto` seems quite unnecessary at this time. Perhaps we can add a `auto_detect` field for `NewOptions` struct instead.

Reply:
Tbh, I like my variant a little bit more, but with your way the code becomes a little bit simpler, so let it be :)